pkg/services/user/user_updater.go

```
@@ -43,6 +52,48 @@ func (o *userUpdater) UpdateUser(id uuid.UUID, user *models.User) (*models.User,
 		return nil, verrs, err
 	}
 
+	// If the update was successful and we are deactivating the user,
+	// update the office and admin statuses to match.
+
+	// Check if we are deactivating the user
+	if !user.Active {
+
+		// Check for Office User
+		foundOfficeUser := models.OfficeUser{}
+		filters = []services.QueryFilter{query.NewQueryFilter("user_id", "=", id.String())}
+		err = o.builder.FetchOne(&foundOfficeUser, filters)
+
+		// If we find a matching Office User, update their status
+		if err == nil {
+			payload := adminmessages.OfficeUserUpdatePayload{
+				Active: &user.Active,
+			}
+			_, verrs, err = o.officeUserUpdater.UpdateOfficeUser(foundOfficeUser.ID, &payload)
+
+			if verrs != nil {
```

I don't want to return these errors here, since the user has already been deactivated and that's the most important outcome of this function. Would love any input on these logs, or if there's additional logging/error handling we should do here!

---

hmmm, I was thinking at first that all of these actions should be in the same transaction, but it really does make sense to keep them separate because these updates are just tidying up the database, they're not functional. I just wish there was a way to return a response that was like a "Yes, but..." so the user could see that the main action succeeded, but this action didn't :(

Maybe these logs should be more prominent than `Info` logs? We don't return the error, but I think putting it at the `logger.Error` level is still justified here

---

_`logger.Error` doesn't cause any errors, right?_ 🤔

---

That was my first thought, but once I read through our logging docs, it sounds like we shouldn't use logger.Error for this, and primarily be using logger.Info? https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging

But I'm with you. I want to make sure these are being surfaced in a meaningful/helpful way.

---

Re-reading that section, now I'm not sure. It _would_ be a failed db update, and it wouldn't be because of a bad id at that point. Perhaps error is ok?

---

yeah, I wasn't sure after reading that either! My interpretation is that `Error` should be used for failures that weren't caused by user error, which applies in this case. This would be a strange and unexpected system failure, so we'd probably want to know so we could find the cause before it affected something more critical. I would defend it in logging court

---

DONE! I will be your co-defense.